payments-core/src/main/java/com/stripe/android/model/ConfirmationToken.kt

```diff
@@ -0,0 +1,122 @@
+
+    /**
+     * Has the value `true` if the object exists in live mode or the value `false` if the object exists in test mode.
+     */
+    @JvmField val liveMode: Boolean,
+
+    /**
+     * Mandate data for this confirmation token. This is automatically generated based on
+     * the payment method and usage, eliminating the need for manual mandate handling.
+     * Internal field containing auto-generated mandate configuration.
+     */
+    @JvmField val mandateData: MandateData?,
+
+    /**
+     * ID of the PaymentIntent that this ConfirmationToken was used to confirm,
+     * or null if this ConfirmationToken has not yet been used.
+     */
+    @JvmField val paymentIntentId: String?,
+
+    /**
+     * Payment method data collected from Elements. This represents the transactional checkout state,
+     * not a reusable PaymentMethod object.
+     */
+    @JvmField val paymentMethodPreview: PaymentMethodPreview?,
+
+    /**
+     * Return URL used to confirm the intent for redirect-based methods.
+     */
+    @JvmField val returnUrl: String?,
+
+    /**
+     * Indicates how you intend to use the payment method for future payments.
+     * This is automatically determined based on Elements configuration and user input.
+     */
+    @JvmField val setupFutureUsage: ConfirmPaymentIntentParams.SetupFutureUsage?,
+
+    /**
+     * ID of the SetupIntent that this ConfirmationToken was used to confirm,
+     * or null if this ConfirmationToken has not yet been used.
+     */
+    @JvmField val setupIntentId: String?,
+
+    /**
+     * Shipping information collected on this ConfirmationToken.
+     */
+    @JvmField val shipping: ShippingInformation?,
+) : StripeModel {
+    /**
+     * Preview of the payment method data collected from Elements.
+     * This represents the transactional checkout state, not a reusable PaymentMethod object.
+     */
+    @Parcelize
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    // TODO(cttsai-stripe): should convert this data class to @Poko class in next major release
+    data class PaymentMethodPreview internal constructor(
+        /**
+         *  This field indicates whether this payment method can be shown again to its customer in a checkout flow.
+         *  Stripe products such as Checkout and Elements use this field to determine
+         *  whether a payment method can be shown as a saved payment method in a checkout flow.
+         *  The field defaults to “unspecified”.
+         */
+        @JvmField val allowRedisplay: PaymentMethod.AllowRedisplay? = null,
+
+        /**
+         *  Billing information associated with the PaymentMethod that may be used or required by
+         *  particular types of payment methods.
+         */
+        @JvmField val billingDetails: PaymentMethod.BillingDetails? = null,
+
+        /**
+         *  The ID of the Customer to which this PaymentMethod is saved.
+         *  This will not be set when the PaymentMethod has not been saved to a Customer.
```

Comment:
Seems worth it to remove the double negative.

```suggestion
         *  This will only be set when the PaymentMethod has been saved to a Customer.
```